### PR TITLE
[wasm] Add --regression suppport for wasm.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/mini/driver.c [hunk 1]
```diff
@@ -350,6 +351,92 @@ domain_dump_native_code (MonoDomain *domain) {
...
+
+	if (strncmp (method->name, "test_", 5) != 0)
+		return FALSE;
+
+	if (interp) {
+		static gboolean filter_method_init = FALSE;
+		static const char *filter_method = NULL;
+
+		if (!filter_method_init) {
+			filter_method = g_getenv ("INTERP_FILTER_METHOD");
```
Comment (lewurm) [inline mono/mini/driver.c]: (can be done later) we should rename it to `REGRESSION_FILTER_METHOD`. it's not really interpreter specific

Comment (kumpera) [reply mono/mini/driver.c]: notes! I'll address that as I add support for running individual tests with wasm.

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

File: mono/mini/gshared.cs [hunk 2]
```diff
@@ -1350,6 +1350,8 @@ class Class6 : IFace6 {
 
 	// FIXME: The runtime-invoke wrapper used by BeginInvoke is not found
 	[Category ("!FULLAOT")]
+	// FIXME: Wasm is single threaded and can't handle blocking waits
+	[Category ("!WASM")]
```
Comment (jaykrell) [inline mono/mini/gshared.cs]: Another time/never: Should be a characteristic, not a specific platform: `[Category ("Threads")]` or Multithreaded or  HasBlockingWaits, etc.

File: mono/mini/driver.c [hunk 3]
```diff
@@ -411,7 +411,7 @@ method_should_be_regression_tested (MonoMethod *method, gboolean interp)
 		if (!is_ok (error))
 			continue;
 
-		char *utf8_str = typed_args[0]; //this points into image memory that is constant
+		char *utf8_str = (char*)(void*)typed_args[0]; //this points into image memory that is constant
```
Comment (jaykrell) [inline mono/mini/driver.c]: No need for the `(void*)` but ok.

**Pull request merged**